docs/src/tutorial-kinds-of-failure.md

```diff
@@ -86,33 +86,23 @@ How does Kani's output change, compared to the unsafe operation?
 <details>
 <summary>Click to see explanation for exercise 1</summary>
 
-Having switched back to the safe indexing operation, Kani reports two failures:
+Having switched back to the safe indexing operation, Kani reports a bounds check failure:
```

Comment:
It may be worth replacing this example. When I run `cargo test` with the unsafe code line
`return unsafe { *a.get_unchecked(i % a.len() + 1) };`
it now fails with this helpful error message:
```
running 2 tests
thread 'bounds_check::tests::doesnt_crash' panicked at library/core/src/[panicking.rs:221:5](http://panicking.rs:221:5):
unsafe precondition(s) violated: slice::get_unchecked requires that the index is within the slice
thread caused non-unwinding panic. aborting.
error: test failed, to rerun pass --lib
```
whereas Kani's error message, instead of saying
```
Failed Checks: dereference failure: pointer outside object bounds
```
as desired, now says:
```
SUMMARY:
 ** 1 of 345 failed (8 unreachable)
Failed Checks: assumption failed
 File: "/Users/jaisnan/.rustup/toolchains/nightly-2024-07-01-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/index.rs", line 273, in <usize as std::slice::SliceIndex<[u32]>>::get_unchecked

VERIFICATION:- FAILED
```
I tried a few other unsafe slice functions and found that `split_at_unchecked` produces a more informative error message from Kani:
```
SUMMARY:
 ** 2 of 343 failed (8 unreachable)
Failed Checks: attempt to compute `unchecked_sub` which would overflow
 File: "/Users/jaisnan/.rustup/toolchains/nightly-2024-07-01-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/mod.rs", line 1985, in core::slice::<impl [u32]>::split_at_unchecked
Failed Checks: dereference failure: pointer outside object bounds
 File: "src/bounds_check.rs", line 15, in bounds_check::get_wrapped

VERIFICATION:- FAILED
```
but `cargo test` catches the error as well. I've held off on replacing it for now in case we want to try to find an example where `cargo test` fails _and_ Kani produces a good error message. I'm happy to do that, but would benefit from some guidance on where to look.

Reply:
I see https://github.com/model-checking/kani/issues/2681 show up here as well :(. Might need to prioritize it sooner rather than later.

As per your example, we'd want an example where `cargo test` passes, while Kani fails with a good error message right?

Reply:
Given the nice error messages, it might be time to revisit: https://github.com/model-checking/kani/issues/1740

Reply:
We might also want to enable `ub_checks` if they are not enabled yet.

Reply:
> If you are replacing what the output message in the tutorial is, can you cut the file path in the summary to only show the relative path or the file name?
>
Yes, will do.
> About your example, we'd want an example where `cargo test` passes, while Kani fails with a good error message right? Not an example where `cargo test` fails?
>
Yes, you're right--I meant that we want to find an example where `cargo test` fails to find a problem, but poor wording on my part.

Reply:
Resolving -- replaced with new example per @zhassan-aws suggestion.